### ISSUE TITLE
False positive for 'Placeholder %q is missing in the translation'

### ISSUE THREAD

**Issue by rillig** [vslavik/poedit#777]
~~~po
# c-format
msgid "Generating key (%d%% done)."
msgstr "Schlüssel wird generiert (%d %% erledigt).\n"
~~~

The inspection warns:
> Platzhalter "%d%" fehlt in der Übersetzung.

This warning sounds as if `%d%` were interpreted as a single placeholder.
Since the message is `c-format`, there are two placeholders, the first being `%d` and the second being `%%`.
Both placeholders already occur in the translated message, with a space between since that is usual in German.

**Comment by vslavik** [comment]
> This warning sounds as if %d% were interpreted as a single placeholder.

Yes, [it is](https://github.com/vslavik/poedit/blob/fd41a83ad401228e7278e5e3615d276a90dce14a/src/syntaxhighlighter.cpp#L195) - Poedit highlights (and checks, because it's the same code) commonly used placeholders too, and`%foo%` is one of them. They're not tagged with any `*-format` tag in practice (and GNU gettext would strip any custom tags off in processing anyway), so the detection is applied indiscriminately.

In highlighting, this probably doesn't show up, because `c-format` has higher priority and repaints it correctly. In placeholder checks both are processed and cause this.

Thanks for alerting me to the issue!
